File: bot/src/main/java/spread/SpreadCalculator.java
package spread;

import parser.FormulaParser;

import java.util.Map;

public class SpreadCalculator {
    private final FormulaParser formulaParser;

    public SpreadCalculator(FormulaParser formulaParser) {
        this.formulaParser = formulaParser;
    }

    public double calculate(String formula, Map<String, Double> prices) {
        return formulaParser.calculate(formula, prices);
    }
}

File: Desktop/python_bot/spread/spread_calculator.py
from __future__ import annotations

from typing import Dict

from parser.formula_parser import FormulaParser


class SpreadCalculator:
    def __init__(self, parser: FormulaParser | None = None) -> None:
        self._parser = parser or FormulaParser()

    def evaluate(self, formula: str, prices: Dict[str, float]) -> float:
        return self._parser.calculate(formula=formula, prices=prices)

